src/GovukNotify/Client/NotificationClient.cs

```diff
@@ -245,6 +245,20 @@ public async Task<byte[]> GetPdfForLetterAsync(string notificationId)
             return response;
         }
 
+        public static JObject PrepareUpload(byte[] documentContents, bool isCsv, bool confirmEmailBeforeDownload, string retentionPeriod)
+        {
+            if (documentContents.Length > 2 * 1024 * 1024) {
+                throw new System.ArgumentException("File is larger than 2MB");
+            }
+            return new JObject
+            {
+                {"file", System.Convert.ToBase64String(documentContents)},
+                {"is_csv", isCsv},
+                {"confirm_email_before_download", confirmEmailBeforeDownload},
+                {"retention_period", retentionPeriod}
+            };
+        }
+
         public static JObject PrepareUpload(byte[] documentContents, bool isCsv = false)
```

Comment:
i don't have massive thoughts on this since i'm not a dotnet expert at all or even an intermediate or even really a beginner, just wondering if it's easier/better to have one function with defaults rather than overloading?

(weak opinion weakly held)

Reply:
That was definitely my preference but in my googling, as a novice c#/.net person too, I couldn't find a way to have optional parameters with a default value of null, so think we have to use overloading instead to get the null values.